packages/protocol-kit/src/SafeProvider.ts

```diff
@@ -23,24 +30,34 @@ import {
   getMultiSendContractInstance,
   getSafeContractInstance,
   getSafeProxyFactoryContractInstance,
+  getSafeWebAuthnSharedSignerContractInstance,
+  getSafeWebAuthnSignerFactoryContractInstance,
   getSignMessageLibContractInstance,
   getSimulateTxAccessorContractInstance
 } from './contracts/contractInstances'
 import {
   SafeProviderTransaction,
   GetContractProps,
   SafeProviderConfig,
-  Eip1193Provider,
-  HttpTransport,
-  SocketTransport
+  SafeSigner,
+  SafeConfig,
+  ContractNetworksConfig
 } from '@safe-global/protocol-kit/types'
+import PasskeySigner from './utils/passkeys/PasskeySigner'
+import { DEFAULT_SAFE_VERSION } from './contracts/config'
 
 class SafeProvider {
   #externalProvider: BrowserProvider | JsonRpcProvider
-  signer?: string
-  provider: Eip1193Provider | HttpTransport | SocketTransport
-
-  constructor({ provider, signer }: SafeProviderConfig) {
+  provider: SafeProviderConfig['provider']
+  signer?: SafeSigner
```

Comment:
I would use the SafeProviderConfig['signer'] in all places and remove the SafeSigner type or add a new SafeEthereumProvider (for example) instead so we would use:

```
provider: SafeProviderConfig['provider'],
signer: SafeProviderConfig['signer']
```

or

```
provider: SafeEthereumProvider,
signer: SafeSigner
```

Reply:
This is only to make it cleaner as this two props are used in multiple places